FirebaseAdmin/FirebaseAdmin.Tests/Messaging/InstanceIdClientTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Http;
+using Xunit;
+
+namespace FirebaseAdmin.Tests.Messaging
+{
+    public class InstanceIdClientTest
+    {
+        private static readonly GoogleCredential MockCredential =
+            GoogleCredential.FromAccessToken("test-token");
+
+        [Fact]
+        public void InstanceIdClientThrowsOnNoProjectId()
+        {
+            var clientFactory = new HttpClientFactory();
+            Assert.Throws<ArgumentException>(
+                () => new InstanceIdClient(clientFactory, MockCredential, null));
+            Assert.Throws<ArgumentException>(
+                () => new InstanceIdClient(clientFactory, MockCredential, string.Empty));
+        }
+
+        [Fact]
+        public void InstanceIdClientThrowsOnNoCredential()
```

Comment:
Shorten the names by dropping redundant labels -- "NoCredential", "NoClientFactory", "SubscribeToTopic" etc